src/lib/encryption.ts

```diff
@@ -38,11 +40,13 @@ function encrypt(message: Field[], otherPublicKey: PublicKey) {
 
   // encryption
   let cipherText = [];
-  for (let messageChunk of message) {
+  for (let i = 0; i < message.length; i++) {
     let keyStream = sponge.squeeze();
-    let encryptedChunk = messageChunk.add(keyStream);
-    sponge.absorb(encryptedChunk); // absorb for the auth tag
+    let encryptedChunk = message[i].add(keyStream);
     cipherText.push(encryptedChunk);
+    // absorb for the auth tag (two at a time for saving permutations)
+    if (i % 2 === 1) sponge.absorb(cipherText[i - 1]);
+    if (i % 2 === 1 || i === message.length - 1) sponge.absorb(cipherText[i]);
```

Comment:
a bit hard to parse, but I'm not sure how you could make it easier to parse. LGTM though